tests/Benchmarks/API/HookBench.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Benchmarks\API;
+
+class HookBench
+{
+    /**
+     * @Revs(1000)
+     * @Iterations(10)
+     * @OutputTimeUnit("microseconds")
+     * @RetryThreshold(10.0)
+     */
+    public function benchWithoutHook()
+    {
+        for ($i = 0; $i < 1000; $i++) {
+            $this->dummyFunction();
+        }
+    }
+
+    public function setUp()
+    {
+        \DDTrace\trace_method('Benchmarks\API\HookBench', 'dummyFunction', function () { });
```

Comment:
Are there differences between `trace_method`, `trace_function` and `install_hook`? Should we test all 3?

Reply:
Since `trace_method` and `trace_function` use nearly the same function under the hook, I'd expect them to be very similar. Although `install_hook` doesn't use the same one, it should be in the same ballpark.

Anyway, this is a great idea, and it does not harm to compare them (considering I don't expect benchmark instability on these API calls).

Reply:
Added the benchmarks. It appears that `install_hook` is non-negligeably slower, especially considering the latter _doesn't open a span_ 🤔 

```
\Benchmarks\API\HookBench

    benchWithoutHook........................I9 - Mo113.253μs (±2.99%)
    benchHookOverheadTraceMethod............I9 - Mo463.337μs (±1.77%)
    benchHookOverheadFunctionTraceFunction..R1 I4 - Mo452.150μs (±2.14%)
    benchHookOverheadInstallHookOnMethod....R1 I9 - Mo1,271.603μs (±2.30%)
    benchHookOverheadInstallHookOnFunction..R1 I9 - Mo1,235.958μs (±2.85%)
```

Reply:
Hmm very interesting. Thanks for adding those. We shall look into it indeed then, especially as we have added it in more integrations recently.